### PR TITLE
Support float zooms in bounds() function

### PR DESCRIPTION
**Pull request by timiyay**
This PR adds support for float zooms (i.e 16.5) in the bounds() function, satisfying issue https://github.com/mapbox/geo-viewport/issues/5.

The real work was done upstream in https://github.com/mapbox/sphericalmercator, and this PR only required a small amount of change:

• adding a test for float-based zoom levels
• updating `node-sphericalmercator` from v1.0.2 to v1.1.0 (latest at time of writing)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
   "bugs": {
     "url": "https://github.com/mapbox/geo-viewport/issues"
-  }, 
-  "version": "0.2.2", 
+  },
+  "version": "0.2.2",
   "dependencies": {
-    "@mapbox/sphericalmercator": "~1.0.2"
-  }, 
+    "@mapbox/sphericalmercator": "~1.1.0"
```
Comment (timiyay) [inline package.json]: This is the meat of the PR - updating a dependency. Easy peasy!

Unfortunately, the diff is a bit noisy for the `package.json`. I'm developing on npm 6, and it insisted on removing the trailing whitespace from every line in the file. Nonetheless, I left it in, as it isn't a difficult change to follow.

**Comment by timiyay** [comment]
@mapsam I've had a crack at adding float zoom support to `bounds()`. It's quite a small change. Whenever you have time and inclination, I'd love a review!

File: test/viewport.js [hunk 2]
```diff
@@ -60,9 +60,24 @@ test('bounds for 512px tiles', function (t) {
...
-    t.equal(precisionRound(xMax, decDegreesFloatTolerance), precisionRound(-77.03365981578827, decDegreesFloatTolerance));
-    t.equal(precisionRound(yMax, decDegreesFloatTolerance), precisionRound(38.89843950894583, decDegreesFloatTolerance));
+    t.equal(precisionRound(xMin, decDegreesFloatTolerance), -77.03945339);
+    t.equal(precisionRound(yMin, decDegreesFloatTolerance), 38.89697827);
+    t.equal(precisionRound(xMax, decDegreesFloatTolerance), -77.03365982);
+    t.equal(precisionRound(yMax, decDegreesFloatTolerance), 38.89843951);
     t.end();
 });
+
+test('bounds for float zooms', function (t) {
```
Comment (timiyay) [inline test/viewport.js]: If you're wondering where the magic values came from, in these test expectations...

I didn't have the necessary background in the `node-sphericalmercator` math to create my expectations with math, so I fell back to good old GIS. 

My original use case for this module was creating GeoJSON bounding boxes that bound Mapbox Static Map API images. So, I used this as my reality check. I'd use the API to request an image, use `bounds()` to determine its bbox, then use QGIS to do a visual check.

To that end, it's slightly imprecise, but matches my use case.

Comment (mapsam) [reply test/viewport.js]: I think this is fine. One thing to do would be add a test for z `16` and z `16.5` and assert that the values are less/greater than each other, essentially asserting they output _different_ values.

No need to do this now, though!

**Pull request merged**